lib/ansible/plugins/inventory/docker_machine.py

```diff
@@ -0,0 +1,159 @@
+keyed_groups:
+  - prefix: tag
+    key: 'dm_tags'
+
+# Example using tag splitting where the tag is like 'dm_tag_gantry_component:routinator'
+strict: False
+split_tags: True
+split_separator: ":"
+keyed_groups:
+  - prefix: gantry_component
+    key: 'dm_tag_gantry_component'
+'''
+
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_native
+from ansible.module_utils._text import to_text
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+
+import json
+import re
+import subprocess
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+    ''' Host inventory parser for ansible using Docker machine as source. '''
+
+    NAME = 'docker_machine'
+
+    def _run_command(self, *args):
+        return subprocess.check_output(["docker-machine"] + list(args)).decode('utf-8').strip()
+
+    def _populate(self):
+        self.inventory.add_group('all')
+
+        try:
+            self.nodes = self._run_command('ls', '-q').splitlines()
+            for self.node in self.nodes:
+                self.node_attrs = json.loads(self._run_command('inspect', self.node))
+
+                id = self.node_attrs['Driver']['MachineName']
+                self.inventory.add_host(id)
+                self.inventory.add_group(self.node_attrs['DriverName'])
+                self.inventory.add_host(id, group=self.node_attrs['DriverName'])
+                self.inventory.add_host(id, group='all')
+
+                # Find out more about the following variables at: https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html
+                self.inventory.set_variable(id, 'ansible_host', self.node_attrs['Driver']['IPAddress'])
+                self.inventory.set_variable(id, 'ansible_port', self.node_attrs['Driver']['SSHPort'])
+                self.inventory.set_variable(id, 'ansible_user', self.node_attrs['Driver']['SSHUser'])
+                self.inventory.set_variable(id, 'ansible_ssh_common_args', '-o StrictHostKeyChecking=no')
+                self.inventory.set_variable(id, 'ansible_ssh_private_key_file', self.node_attrs['Driver']['SSHKeyPath'])
+
+                # pass '--shell=bash' to workaround 'Error: Unknown shell
+                env_out = self._run_command('env', '--shell=bash', id)
+
+                # example output of docker-machine env
+                #   export DOCKER_TLS_VERIFY="1"
+                #   export DOCKER_HOST="tcp://134.209.204.160:2376"
+                #   export DOCKER_CERT_PATH="/root/.docker/machine/machines/routinator"
+                #   export DOCKER_MACHINE_NAME="routinator"
+                #   # Run this command to configure your shell:
+                #   # eval $(docker-machine env --shell=bash routinator)
+
+                for env_var_name in ['DOCKER_TLS_VERIFY', 'DOCKER_HOST', 'DOCKER_CERT_PATH', 'DOCKER_MACHINE_NAME']:
+                    env_var_value = re.search('{0}="([^"]+)"'.format(env_var_name), env_out).group(1)
```

Comment:
You should allow for some of these env variables not being available (such as the TLS related ones). Maybe:
```suggestion
                    matcher = re.search('{0}="([^"]+)"'.format(env_var_name), env_out)
                    if matcher:
                        env_var_value = matcher.group(1)
                        self.inventory.set_variable(id, ...
```

Reply:
I assumed that all four would always be generated by `docker-machine env` as a complete set with none missing. However, I was also thinking that for future proofing the code should perhaps not expect specific vars but just capture all that are output by `docker-machine env`.

Reply:
If it only passes them on anyway, it's probably better to catch them all. I just hope that they don't use fancy line-break support / escaping support. It would really be nice if there would be an easy machine-readable JSON export or something like that...

Reply:
I pushed a commit which changes this to capture whatever DOCKER_xxx env vars are output by `docker-machine env`, rather than hard-code the vars it is looking for.